portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,138 @@
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    List<Task> tasks = new ArrayList<>();                  
+    
+    for (Entity entity : results.asIterable()) {
+    //for( int i = 0; i < numComments ; i++) {
+      long id = entity.getKey().getId();
+      String name = (String) entity.getProperty("name");
+      String comments = (String) entity.getProperty("comments");
+      long timestamp = (long) entity.getProperty("timestamp");
+     
+      Task task = new Task(id, name, comments, timestamp);
+      tasks.add(task);
+    }
+    
+    Gson gson = new Gson();
+    
+    response.setContentType("application/json;");
+    //String json = convertToJsonUsingGson(messages);
+    response.getWriter().println(gson.toJson(tasks));
+    /*
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("<h1>Hello Niha!</h1>");
+    */
+
+    // Create Arraylist of hard coded messages
+    
+    //messages.add("My favorite foods include dolmas, dark chocolate, and apricots.");
+    //messages.add("My favorite National Park is Denali National Park in Alaska.");
+    //messages.add("My favorite yoga styles include Vinyasa, Yin, and Iyengar.");
+
+    // Convert the server stats to JSON
+    // Send the JSON as the response
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
+    String comments = request.getParameter("comments");
+    long timestamp = System.currentTimeMillis();
+    
+    numComments = getPlayerChoice(request);
+    
+    if (numComments == -1) {
+      response.sendRedirect("/index.html");
+      response.getWriter().println("Please enter an integer greater than 0.");
+      return;
+    }
+
+
+    //messages.add(name + "," + comments);
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("name", name);
+    taskEntity.setProperty("comments", comments);
+    taskEntity.setProperty("timestamp", timestamp);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+
+    response.sendRedirect("/index.html");
   }
+
+  /**
+   * Converts a ServerStats instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJsonUsingGson(ArrayList<String> messages) {
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+    return json;
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
```

Comment:
(nitpick) Use the ternary operator here: `return value == null ? defaultValue : value;`